controller/array_action/array_mediator_svc.py

```diff
@@ -764,6 +764,65 @@ def get_host_by_host_identifiers(self, initiators):
             raise array_errors.MultipleHostsFoundError(initiators, fc_host)
         return host_name, list(connectivity_types)
 
+    def _get_host_names_by_wwpn(self, host_wwpn):
+        fabrics = self._lsfabric(wwpn=host_wwpn).as_list
+        return set(fabric.name for fabric in fabrics)
+
+    def _lsnvmefabric(self, host_nqn):
+        try:
+            return self.client.svcinfo.lsnvmefabric(remotenqn=host_nqn).as_list
+        except(svc_errors.CommandExecutionError, CLIFailureError) as ex:
+            logger.error("Failed to get nvme fabrics. Reason "
+                         "is: {0}".format(ex))
+            raise ex
+
+    def _get_host_names_by_nqn(self, nqn):
+        nvme_fabrics = self._lsnvmefabric(nqn)
+        return set(nvme_fabric.object_name for nvme_fabric in nvme_fabrics)
+
+    def _lshostiplogin(self, iqn):
+        try:
+            return self.client.svcinfo.lshostiplogin(object_id=iqn).as_single_element
+        except(svc_errors.CommandExecutionError, CLIFailureError) as ex:
+            if SPECIFIED_OBJ_NOT_EXIST in ex.my_message:
+                return None
+            logger.error("Failed to get iscsi host. Reason "
+                         "is: {0}".format(ex))
+            raise ex
+
+    def _get_host_name_by_iqn(self, iqn):
+        iscsi_login = self._lshostiplogin(iqn)
+        if iscsi_login:
+            return iscsi_login.host_name
+        return None
+
+    def get_host_by_host_identifiers(self, initiators):
+        logger.debug("Getting host name for initiators : {0}".format(initiators))
+        host_names = set()
+        connectivity_types = set()
```

Comment:
why do we need to use it? and add to it a value in the if statements? 
I think we just can use the `connectivity_type` value, because if it's `iscsi`, `nvme` or `fc`, so it will be good and if it isn't one of them, we won't use this value anyway after the if statments

Reply:
> we won't use this value anyway after the if statments

we return `connectivity_types`, because according to the current mediator interface, this method returns all the connectivity types it found for the host definition.
so I'm not sure what you mean

Reply:
never mind my bad I missed